synse/commands/fan_sensors.py

```diff
@@ -24,23 +72,42 @@
     #
     # to read all the devices of a given type or model.
 
+    start_time = datetime.datetime.now()
     _cache = await cache.get_metainfo_cache()
+    scan_cache = await cache.get_scan_cache()
 
     readings = []
+    new_readings = dict()
+    new_readings['racks'] = OrderedDict()
 
     logger.debug('--- FAN SENSORS start ---')
     for _, v in _cache.items():
 
         logger.debug('FAN SENSORS')
-        logger.info('fan_sensors cache item: {}'.format(v))
+        logger.debug('fan_sensors cache item: {}'.format(v))
 
         is_temp = v.type.lower() == 'temperature' and v.model.lower() == 'max11610'
         is_pressure = v.type.lower() == 'pressure' and v.model.lower() == 'sdp610'
 
         if is_temp or is_pressure:
-            rack = v.location.rack
-            board = v.location.board
-            device = v.uid
+            rack = v.location.rack # string (vec1-c1-wrigley for example)
+            board = v.location.board # string (vec for example)
+            device = v.uid # string (uuid - only unique to one rack)
+
+            # Find the device in the scan_cache.
+            scan_cache_board = None
+            scan_cache_device = None
+            scan_cache_rack = next((r for r in scan_cache['racks'] if r['id'] == rack), None)
+            if scan_cache_rack is not None:
+                scan_cache_board = next(
+                    (b for b in scan_cache_rack['boards'] if b['id'] == board), None)
+                if scan_cache_board is not None:
+                    scan_cache_device = next(
+                        (d for d in scan_cache_board['devices'] if d['id'] == device), None)
+            logger.debug('scan_cache_rack_id, board_id, device_info: {}, {}, {}'.format(
+                scan_cache_rack.get('id', None),
+                scan_cache_board.get('id', None),
+                scan_cache_device.get('info', None)))
```

Comment:
I'm a little confused about what exactly is going on here. 

we're iterating through the `_cache` which is the metainfo cache. these bits are then traversing through the scan cache, but the scan cache is built off of the metainfo cache, so when we are searching for a rack, board, device, info, etc in the scan cache, we should already be able to access it via the loop variable `v` (bad name for it now that things have gotten more complex.. `v` is really the device, specifically the MetainfoResponse for the device - as defined in the gRPC proto: https://github.com/vapor-ware/synse-server-grpc/blob/master/synse.proto#L166)

so really, `scan_cache_board` should always be the same as `board`, `scan_cache_device` should be the same as `device`, and `scan_cache_rack` should be the same as `rack`.

Reply:
In the initial result set I just pasted in above, there is no device_info. Auto fan needs the mapping to the device_info for sanity's sake.

Without it, guess where the rack location is for this thermistor? I don't know either: c5d8272bc1106aa12225850d33483f33

Reply:
Yes - I'm essentially joining tables.

Reply:
Each data point is a join of these data (reading, location, and scan)
There are currently 54 thermistor and 18 pressure sensor reads in the result set. 18 more thermistors are coming in hardware.
``
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":19.67,
        "timestamp":"2018-03-27 21:48:07.742104699 +0000 UTC m=+96.137714757",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"1e93da83dd383757474f539314446c3d"
    },
    "scan_cache_device":{
      "id":"1e93da83dd383757474f539314446c3d",
      "info":"Rack Temperature 3 Spare",
      "type":"temperature"
    }
  },
```